hypervisor/arch/x86/guest/vm.c

```diff
@@ -57,16 +57,19 @@ void *get_sworld_memory_base(void)
 	return post_user_vm_sworld_memory;
 }
 
-uint16_t get_vmid_by_uuid(const uint8_t *uuid)
+uint16_t get_vmid_by_name(const char *name)
 {
-	uint16_t vm_id = 0U;
-
-	while (!vm_has_matched_uuid(vm_id, uuid)) {
-		vm_id++;
-		if (vm_id == CONFIG_MAX_VM_NUM) {
-			break;
+	uint16_t vm_id = ACRN_INVALID_VMID;
+	uint16_t matched_cnt = 0;
+
+	/* check if there are duplicate VM names in vm configurations */
+	for (uint16_t idx = 0U; idx < CONFIG_MAX_VM_NUM; idx++) {
+		if ((*name != '\0') && vm_has_matched_name(idx, name)) {
+			matched_cnt++;
+			vm_id = (matched_cnt > 1) ? (ACRN_INVALID_VMID) : (idx);
```

Comment:
counting the matched name is not necessary,

We call it get_vmid_by_name, we can return the first matching vm_id.

Reply:
this API is also used by post-launched VM, for pre-launched VM, check  get_vmid_by_name != vm_id is ok,
for post-launched VM, the vmid is unknow before call  get_vmid_by_name